src/main/java/com/moneytransfer/service/CurrencyExchangeServiceImpl.java

```diff
@@ -23,14 +23,23 @@ public class CurrencyExchangeServiceImpl implements CurrencyExchangeService {
     @Value("${freecurrencyapi.apiUrl}")
     private String apiUrl;
 
+    /**
+     * Performs the currency exchange.
+     *
+     * @param amount
+     * @param sourceCurrency
+     * @param targetCurrency
+     * @return
+     * @throws MoneyTransferException
+     */
     public BigDecimal exchangeCurrency(BigDecimal amount, final Currency sourceCurrency, final Currency targetCurrency) throws MoneyTransferException {
         var subUrl = apiUrl.concat(apiKey);
         var url = String.format("%1$s&currencies=%2$s&base_currency=%3$s", subUrl, targetCurrency.name(), sourceCurrency.name());
         ResponseEntity<ExchangeRatesResponse> responseEntity = new RestTemplate().exchange(url, HttpMethod.GET, null, ExchangeRatesResponse.class);
```

Comment:
A couple notes here:
1. RestTemplate is thread safe, therefore you can have 1 instance here. 
2. If I remember correctly, restTemplate has a get method.

Reply:
1. ok, instantiated RestTemplate outside the method.
2. Thanks, used getForEntity instead.